core/server/data/xml/sitemap/base-generator.js

```diff
@@ -55,13 +55,18 @@ _.extend(BaseSiteMapGenerator.prototype, {
         // Create all the url elements in JSON
         var self = this,
             nodes;
-        nodes = _.map(data, function (datum) {
+
+        nodes = _.reduce(data, function (nodeArr, datum) {
```

Comment:
question:
i wonder why we need the changes from line 59 till 69, because when asking for `getData`, this function should already return filtered data based on the query.